src/other/mlpmixer.jl

```diff
@@ -1,83 +1,106 @@
+                                    DropPath(drop_path_rate)), +))
+end
+
+"""
+    mlpmixer(block, imsize::NTuple{2} = (224, 224); inchannels = 3, norm_layer = LayerNorm,
+             patch_size::NTuple{2} = (16, 16), embedplanes = 512, depth = 12, 
+             mlp_ratio = (0.5, 4.0), mlp_layer = mlp_block, dropout = 0., drop_path_rate = 0., 
+             activation = gelu, nclasses = 1000)
 
 Creates a model with the MLPMixer architecture.
 ([reference](https://arxiv.org/pdf/2105.01601)).
 
 # Arguments
+- `block`: the type of mixer block to use in the model - architecture dependent
 - `imsize`: the size of the input image
 - `inchannels`: the number of input channels
+- `norm_layer`: the normalization layer to use in the model
 - `patch_size`: the size of the patches
-- `planes`: the number of channels fed into the main model
-- `depth`: the number of blocks in the main model
-- `expansion_factor`: the number of channels in each block
-- `dropout`: the dropout rate
-- `nclasses`: the number of classes in the output
-- `token_mix`: the function to use for the token mixing layer
-- `channel_mix`: the function to use for the channel mixing layer
-"""
-function mlpmixer(imsize::NTuple{2} = (256, 256); inchannels = 3, patch_size = 16, planes = 512, 
-                  depth = 12, expansion_factor = 4, dropout = 0., nclasses = 1000, token_mix = 
-                  _conv1d, channel_mix = Dense)
-                    
-  im_height, im_width = imsize
-
-  @assert (im_height % patch_size) == 0 && (im_width % patch_size == 0)
-  "image size must be divisible by patch size"
-  
-  num_patches = (im_height ÷ patch_size) * (im_width ÷ patch_size)
-
-  layers = []
-  push!(layers, PatchEmbedding(patch_size))
-  push!(layers, Dense((patch_size ^ 2) * inchannels, planes))
-  append!(layers, [Chain(_residualprenorm(planes, mlp_block(num_patches, 
-                                          expansion_factor * num_patches; 
-                                          dropout, dense = token_mix)),
-                         _residualprenorm(planes, mlp_block(planes, 
-                                          expansion_factor * planes; dropout, 
-                                          dense = channel_mix)),) for _ in 1:depth])
-
-  classification_head = Chain(_seconddimmean, Dense(planes, nclasses))
-
-  return Chain(Chain(layers...), classification_head)
+- `embedplanes`: the number of channels after the patch embedding (denotes the hidden dimension)
+- `depth`: the number of blocks in the model
+- `mlp_ratio`: number(s) that determine(s) the number of hidden channels in the token mixing MLP 
+               and/or the channel mixing MLP as a ratio to the number of planes in the block.
+- `mlp_layer`: the MLP block to use
+- `dropout`: the dropout rate to use in the MLP blocks
+- `drop_path_rate`: Stochastic depth rate
+- `activation`: the activation function to use in the MLP blocks
```

Comment:
This makes sense, but we lose default value information, do we not?

Reply:
We can still document the default set for the default value of `block` in the docstring. But as it stands, `block` is not really customizable unless it meets a strict set of arguments and keywords.

Reply:
No I mean if we group the keywords how do we specify the default values?

Reply:
Is that not already handled by `mixerblock`? If `block_kwargs...` is empty meaning the user specified nothing, then you pass nothing to `mixerblock` which already has defaults.

This is not the case right now, but if `block` had a keyword argument with no default, but we are able to specify a default 1-level up, then we would use multiple dispatch of the type of `block` to insert that default into the `block_kwargs...`. Something like:
```julia
# use a shadow method
function _mlpmixer(block, ...; block_kwargs...)
 ...
end
mlpmixer(block::typeof(mixerblock), ...; specific_kw = val, kwargs...) = _mlpmixer(block, ...; specific_kw = val, kwargs...)
mlpmixer(block, ...; kwargs...) = _mlpmixer(block, ...; kwargs...)
```

Reply:
This seems more like it - I'll work out a version of this that works with maximum API exposure if the user needs but hiding it completely in case they just want the default options

Reply:
A complete version of this seems near impossible because the values are being used everywhere and exposing the API makes more sense, but I've cleaned it up a bit for now